api/src/main/java/com/cloud/network/Network.java

```diff
@@ -268,6 +268,7 @@ public static class Capability {
         public static final Capability LoadBalancingSupportedIps = new Capability("LoadBalancingSupportedIps");
         public static final Capability AllowDnsSuffixModification = new Capability("AllowDnsSuffixModification");
         public static final Capability RedundantRouter = new Capability("RedundantRouter");
+        public static final Capability SelectSnatIpAllowed = new Capability("SelectSnatIpAllowed");
```

Comment:
The capability name seems inconsistent with other capabilities.
Maybe better to use AllowSpecificSourceNatIp or just SpecifySourceNatIp

Reply:
+1 on SpecifySourceNatIp